eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ChannelOpenDualFunded.scala

```diff
@@ -349,8 +353,144 @@ trait ChannelOpenDualFunded extends DualFundingHandlers with ErrorHandlers {
+      // A better option would be to double-spend the funding transaction(s).
+      log.warning("cannot force-close while dual-funded transactions are unconfirmed")
+      val replyTo = if (c.replyTo == ActorRef.noSender) sender() else c.replyTo
+      replyTo ! RES_FAILURE(c, CommandUnavailableInThisState(d.channelId, "force-close", stateName))
+      stay()
+
+    case Event(e: Error, d: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED) => handleRemoteError(e, d)
+  })
+
+  when(WAIT_FOR_DUAL_FUNDING_READY)(handleExceptions {
+    case Event(channelReady: ChannelReady, d: DATA_WAIT_FOR_DUAL_FUNDING_READY) =>
+      val shortIds1 = d.shortIds.copy(remoteAlias_opt = channelReady.alias_opt)
+      shortIds1.remoteAlias_opt.foreach { remoteAlias =>
+        log.info("received remoteAlias={}", remoteAlias)
+        context.system.eventStream.publish(ShortChannelIdAssigned(self, d.commitments.channelId, shortIds = shortIds1, remoteNodeId = remoteNodeId))
+      }
+      log.info("shortIds: real={} localAlias={} remoteAlias={}", shortIds1.real.toOption.getOrElse("none"), shortIds1.localAlias, shortIds1.remoteAlias_opt.getOrElse("none"))
+      // we create a channel_update early so that we can use it to send payments through this channel, but it won't be propagated to other nodes since the channel is not yet announced
+      val scidForChannelUpdate = Helpers.scidForChannelUpdate(channelAnnouncement_opt = None, shortIds1.localAlias)
+      log.info("using shortChannelId={} for initial channel_update", scidForChannelUpdate)
+      val fees = getRelayFees(nodeParams, remoteNodeId, d.commitments)
+      val initialChannelUpdate = Announcements.makeChannelUpdate(nodeParams.chainHash, nodeParams.privateKey, remoteNodeId, scidForChannelUpdate, nodeParams.channelConf.expiryDelta, d.commitments.remoteParams.htlcMinimum, fees.feeBase, fees.feeProportionalMillionths, d.commitments.capacity.toMilliSatoshi, enable = Helpers.aboveReserve(d.commitments))
+      // we need to periodically re-send channel updates, otherwise channel will be considered stale and get pruned by network
+      context.system.scheduler.scheduleWithFixedDelay(initialDelay = REFRESH_CHANNEL_UPDATE_INTERVAL, delay = REFRESH_CHANNEL_UPDATE_INTERVAL, receiver = self, message = BroadcastChannelUpdate(PeriodicRefresh))
+      // used to get the final shortChannelId, used in announcements (if minDepth >= ANNOUNCEMENTS_MINCONF this event will fire instantly)
+      blockchain ! WatchFundingDeeplyBuried(self, d.commitments.commitInput.outPoint.txid, ANNOUNCEMENTS_MINCONF)
+      goto(NORMAL) using DATA_NORMAL(d.commitments.copy(remoteNextCommitInfo = Right(channelReady.nextPerCommitmentPoint)), shortIds1, None, initialChannelUpdate, None, None, None) storing()
+
+    case Event(_: TxInitRbf, d: DATA_WAIT_FOR_DUAL_FUNDING_READY) =>
+      // Our peer may not have received the funding transaction confirmation.
+      stay() sending TxAbort(d.channelId, InvalidRbfTxConfirmed(d.channelId).getMessage)
```

Comment:
Agreed, those two handlers are really worth factoring, done in https://github.com/ACINQ/eclair/pull/2274/commits/33b60156078b406135955af9dbd130d0556941fe

Reply:
One nit tough. I'd rather use `d1` instead of `nextData`, because the latter is dangerously close to `nextStateData`, which is a method from the FSM class that returns the same type. It's also a convention we have pretty much everywhere.

https://github.com/ACINQ/eclair/blob/33b60156078b406135955af9dbd130d0556941fe/eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ChannelOpenDualFunded.scala#L459

Reply:
Done in https://github.com/ACINQ/eclair/pull/2274/commits/7b7bfac5bb44092cac0a7439723abff63ae4f356